src/ImageSharp/Processing/Transforms/ProjectiveTransformHelper.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Transforms
+{
+    /// <summary>
+    /// Enumerates the various options which determine which side to taper
+    /// </summary>
+    public enum TaperSide
+    {
+        /// <summary>
+        /// Taper the left side
+        /// </summary>
+        Left,
+
+        /// <summary>
+        /// Taper the top side
+        /// </summary>
+        Top,
+
+        /// <summary>
+        /// Taper the right side
+        /// </summary>
+        Right,
+
+        /// <summary>
+        /// Taper the bottom side
+        /// </summary>
+        Bottom
+    }
+
+    /// <summary>
+    /// Enumerates the various options which determine how to taper corners
+    /// </summary>
+    public enum TaperCorner
+    {
+        /// <summary>
+        /// Taper the left or top corner
+        /// </summary>
+        LeftOrTop,
+
+        /// <summary>
+        /// Taper the right or bottom corner
+        /// </summary>
+        RightOrBottom,
+
+        /// <summary>
+        /// Taper the both sets of corners
+        /// </summary>
+        Both
+    }
+
+    /// <summary>
+    /// Provides helper methods for working with generalized projective transforms.
+    /// </summary>
+    public static class ProjectiveTransformHelper
+    {
+        /// <summary>
+        /// Creates a matrix that performs a tapering projective transform.
+        /// <see href="https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/graphics/skiasharp/transforms/non-affine"/>
+        /// </summary>
+        /// <param name="size">The rectangular size of the image being transformed.</param>
+        /// <param name="taperSide">An enumeration that indicates the side of the rectangle that tapers.</param>
+        /// <param name="taperCorner">An enumeration that indicates on which corners to taper the rectangle.</param>
+        /// <param name="taperFraction">The amount to taper.</param>
+        /// <returns>The <see cref="Matrix4x4"/></returns>
+        public static Matrix4x4 CreateTaperMatrix(Size size, TaperSide taperSide, TaperCorner taperCorner, float taperFraction)
```

Comment:
Cool suff! Covers basic cases & good for testing without getting outside the scope of the library.

Reply:
Thanks for adding the extra tests, they affirm the expected output well.